google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/it/env/DirectPathEnv.java

```diff
@@ -0,0 +1,182 @@
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.cloud.bigtable.data.v2.stub.EnhancedBigtableStubSettings;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Test environment that uses an existing bigtable table in the directpath environment.
+ *
+ * <p>The test environment can be accessed via both, CFE and direct path. The transport is
+ * controlled via the environment variable {@code GOOGLE_CLOUD_ENABLE_DIRECT_PATH}. Which contains
+ * comma delimited list of service names that should be accessed via direct path.
+ *
+ * <p>The target table must have a pre-existing family {@code cf}. The target table is configured
+ * via the system properties:
+ *
+ * <ul>
+ *   <li>{@code bigtable.project}
+ *   <li>{@code bigtable.instance}
+ *   <li>{@code bigtable.table}
+ * </ul>
+ */
+public class DirectPathEnv implements TestEnv {
+  // TODO(igorbernstein): move direct path conditional logic to gax
+  private static final String DIRECT_PATH_ENV_VAR = "GOOGLE_CLOUD_ENABLE_DIRECT_PATH";
+  private static final String DIRECT_PATH_SERVICE_NAME =
+      "testdirectpath-bigtable.sandbox.googleapis.com";
+
+  private static final String PROJECT_PROPERTY_NAME = "bigtable.project";
+  private static final String INSTANCE_PROPERTY_NAME = "bigtable.instance";
+  private static final String TABLE_PROPERTY_NAME = "bigtable.table";
+
+  private final boolean directPathEnabled;
+  private final String projectId;
+  private final String instanceId;
+  private final String tableId;
+  private static final String FAMILY_ID = "cf";
+  private String rowPrefix;
+
+  private BigtableDataClient dataClient;
+
+  static DirectPathEnv create() {
+    return new DirectPathEnv(
+        isDirectPathEnabled(),
+        getRequiredProperty(PROJECT_PROPERTY_NAME),
+        getRequiredProperty(INSTANCE_PROPERTY_NAME),
+        getRequiredProperty(TABLE_PROPERTY_NAME));
+  }
+
+  public DirectPathEnv(
+      boolean directPathEnabled, String projectId, String instanceId, String tableId) {
+    this.directPathEnabled = directPathEnabled;
+    this.projectId = projectId;
+    this.instanceId = instanceId;
+    this.tableId = tableId;
+    this.rowPrefix = UUID.randomUUID() + "-";
+  }
+
+  @Override
+  public void start() throws IOException {
+    BigtableDataSettings.Builder settingsBuilder =
+        BigtableDataSettings.newBuilder().setProjectId(projectId).setInstanceId(instanceId);
+
+    // Direct path test environment uses a different endpoint.
+    settingsBuilder.stubSettings().setEndpoint(DIRECT_PATH_SERVICE_NAME + ":443");
```

Comment:
should we extract the port as a variable as well, or combine the two?

Reply:
done